components/forms/form-builder/FormViewBuilder.tsx

```diff
@@ -115,34 +142,158 @@ export default function FormViewBuilder({
   );
 }
 
-function displayListValue(fieldVal: any, fieldType?: string) {
-  if (fieldVal === null || fieldVal === "") return "Not provided";
-  if (fieldType === "array") {
-    if (fieldVal.length === 0) return "Not provided";
-    return fieldVal.map((item: any, index: number) => (
-      <Card key={index} className="container-form-view">
-        <Card.Content>
-          {Object.entries(item).map((entry: any, index: number) => (
-            <Container key={index}>
-              <Row style={{ marginBottom: "0.5em" }}>
-                <Col width="one-half">
-                  <Label>
-                    <b>{formatFieldName(entry[0])}</b>
-                  </Label>
-                </Col>
-                <Col width="one-half">{displayListValue(entry[1])}</Col>
-              </Row>
-            </Container>
-          ))}
-        </Card.Content>
-      </Card>
-    ));
+function formatEntryValue(value: any, fieldType?: string) {
+  if (value === null || value === undefined || value === "")
+    return "Not provided";
+  if (fieldType === "date") {
+    return DateUtilities.ToLocalDate(value);
   }
-  if (fieldVal && (fieldType === "date" || strDateRegex.test(fieldVal))) {
-    return DateUtilities.ToLocalDate(fieldVal);
+  // fallback (e.g. nested type is unknown)
+  if (
+    (!fieldType || fieldType === "string") &&
+    typeof value === "string" &&
+    strDateRegex.test(value)
+  ) {
+    return DateUtilities.ToLocalDate(value);
   }
-  if (typeof fieldVal === "boolean") {
-    return fieldVal ? "Yes" : "No";
+
+  if (typeof value === "boolean") {
+    return value ? "Yes" : "No";
+  }
+  return value.toString();
+}
+
+type ArrayFieldRendererProps = {
+  fieldVal: FormData[];
+  field: Field;
+  canEdit: boolean;
+  pageIndex: number;
+  jsonFormName: FormName;
+};
+
+function ArrayFieldRenderer({
+  fieldVal,
+  field,
+  canEdit,
+  pageIndex,
+  jsonFormName
+}: ArrayFieldRendererProps) {
+  if (!fieldVal || fieldVal.length === 0) {
+    return (
+      <ArrayPanel
+        title={
+          <p data-cy="empty-array-panel-val" style={{ fontSize: "19px" }}>
+            Not provided
+          </p>
+        }
+        action={
+          canEdit ? (
+            <ChangeLink
+              targetField={field.name}
+              label={field.label ?? ""}
+              jsonFormName={jsonFormName}
+              pageIndex={pageIndex}
+            />
+          ) : null
+        }
+      />
+    );
   }
-  return fieldVal?.toString();
+
+  return (
+    <>
+      {fieldVal.map((item: FormData, index: number) => (
+        <ArrayPanel
+          key={index}
```

Comment:
Think we can still use index here because we're not adding/removing panels, just viewing them.